tools/failures.py

```diff
@@ -145,7 +145,7 @@ def prepare_the_database():
     if len(outdate_jobs) > 0:
         print "we have %s outdate jobs need to remove" % len(outdate_jobs)
         for job in outdate_jobs:
-            run_query("delete from seta where jobtype=%s" % job)
+            run_query("delete from seta where jobtype='%s' and date<='%s'" % (job, date))
 
```

Comment:
I keep the outdate_jobs because I think people may want to know we are going to delete data for them. But it has a  little redundance. Maybe we could find another way to fix that?